### DISCUSSION TITLE
Is there a way to timeout code that has entered an infinite loop?

### DISCUSSION THREAD

**Discussion by jorroll** [vitest-dev/vitest#1226] [Q&A] [answered]
Inside a test, some buggy code caused an infinite loop. Is there a way to tell Vitest to timeout synchronous code which takes too long? The config's default [`testTimeout`](https://vitest.dev/config/#testtimeout) seems to have no affect and the problematic test never completes.

**Comment by Yokozuna59** [comment, marked as answer]
I guess you have entered infinite inside a synchronous function similar to what I've done. This is the issue you are looking for, #2920, although there is no update on it.
